staging/src/k8s.io/code-generator/cmd/client-gen/generators/generator_for_type.go

```diff
@@ -462,6 +467,16 @@ func new$.type|publicPlural$(c *$.GroupGoName$$.Version$Client) *$.type|privateP
 var listTemplate = `
 // List takes label and field selectors, and returns the list of $.resultType|publicPlural$ that match those selectors.
 func (c *$.type|privatePlural$) List(ctx context.Context, opts $.ListOptions|raw$) (result *$.resultType|raw$List, err error) {
+    if watchListOptions, hasWatchListOptionsPrepared, watchListOptionsErr := $.PrepareWatchListOptionsFromListOptions|raw$(opts); watchListOptionsErr != nil {
+        return &$.resultType|raw$List{}, watchListOptionsErr
+    } else if hasWatchListOptionsPrepared {
+        result, err = c.watchList(ctx, watchListOptions)
+        if err == nil {
+            $.CheckWatchListFromCacheDataConsistencyIfRequested|raw$(ctx, "watchlist request for $.type|resource$", c.list, opts, result)
+            return
+        }
+        klog.Warningf("The watchlist request for $.type|resource$ ended with an error, falling back to the standard LIST semantics, err = %v", err)
```

Comment:
@deads2k is this fallback mechanism okay with you? This matches what we have in the reflector. If we don't like it, then we need to handle the "server doesn't support the feature" case separately from handling other errors.

Reply:
I much prefer the factoring in https://github.com/kubernetes/kubernetes/pull/124509/files#r1638349460

Reply:
refactored to code to look like the example you showed.

Reply:
+1 to David, thanks!